WalletWasabi/Services/UpdateManager.cs

```diff
@@ -52,7 +52,7 @@ public async Task TryUpdateApplicationAsync(CancellationToken cancellationToken)
 			var updateStatus = await WasabiClient.CheckUpdatesAsync(cancellationToken).ConfigureAwait(false);
 			var result = await GetLatestReleaseFromGithubAsync(cancellationToken).ConfigureAwait(false);
 
-			bool updateAvailable = !updateStatus.BackendCompatible || Helpers.Constants.ClientVersion <= result.LatestClientVersion;
```

Comment:
You mean: I have a backend at `v4` and client updates because `Helpers.Constants.ClientVersion <= result.LatestClientVersion` is `true`, the update causes the client to be `v5` and is not compatible anymore, is that correct?

So this PR doesn't change this behavior. I believe yes it might be a relevant scenario, but I'm not sure how to improve things: a client update can always break compatibility with backend, the old client cannot know if it will or not. 

A backend working only with a custom set of clients can very well fork the app and change the endpoint where to look updates to.

Do you see a way to improve current logic?

Reply:
This code makes no sense to me either.

What does it mean that `!updateStatus.BackendCompatible` is `true` but `Helpers.Constants.ClientVersion <= result.LatestClientVersion` is `false`? We will try to download a new client because of the `||` but try to download what? We just checked and there is no update available...

I would suggest to push this diff, can you verify that it makes sense? Thanks

```diff
diff --git a/WalletWasabi/Services/UpdateManager.cs b/WalletWasabi/Services/UpdateManager.cs
index 02674e6dff..52fd810be2 100644
--- a/WalletWasabi/Services/UpdateManager.cs
+++ b/WalletWasabi/Services/UpdateManager.cs
@@ -50,10 +50,9 @@ public class UpdateManager : PeriodicRunner
        {
                try
                {
-                       var updateStatus = await WasabiClient.CheckUpdatesAsync(cancellationToken).ConfigureAwait(false);
                        var result = await GetLatestReleaseFromGithubAsync(cancellationToken).ConfigureAwait(false);
 
-                       bool updateAvailable = !updateStatus.BackendCompatible || Helpers.Constants.ClientVersion <= result.LatestClientVersion;
+                       bool updateAvailable = Helpers.Constants.ClientVersion <= result.LatestClientVersion;
 
                        if (!updateAvailable)
                        {
@@ -62,6 +61,9 @@ public class UpdateManager : PeriodicRunner
                                return;
                        }
 
+                       // Check whether the update is critical.
+                       var updateStatus = await WasabiClient.CheckUpdatesAsync(cancellationToken).ConfigureAwait(false);
+                       
                        if (DownloadNewVersion)
                        {
                                (string installerPath, Version newVersion) = await GetInstallerAsync(result, cancellationToken).ConfigureAwait(false);
```